### PR TITLE
Add support for passing script type in more methods

### PR DESCRIPTION
**Pull request by aidinabedi**
PR improves the following methods by adding the option to pass the type of a script instead of string name (just like `pc.ScriptComponent#create` and `pc.ScriptRegistry#add`):
- `pc.ScriptComponent#has`
- `pc.ScriptComponent#get`
- `pc.ScriptComponent#destroy`
- `pc.ScriptComponent#swap` (marked as private)
- `pc.ScriptComponent#move`
- `pc.ScriptRegistry#has`
- `pc.ScriptRegistry#remove`

This allows script managing to be more safe and maintainable as well as making the API more unified and consistent.

Note: PR also optimizes the code a bit in both `pc.ScriptComponent#create` and `pc.ScriptRegistry#add`.

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by willeastcott** [comment]
@Maksims Any comments on this one?

**Comment by aidinabedi** [comment]
@willeastcott Fixed. Promise to run tests before submitting next time :)

**Comment by Maksims** [comment]
No objections, nice one.

**Comment by willeastcott** [comment]
Awesome. Thanks @Maksims! Let's meeeeeerge! 😄

File: src/script/script-registry.js [hunk 1]
```diff
@@ -178,19 +185,26 @@ Object.assign(pc, function () {
...
      */
-    ScriptRegistry.prototype.has = function (name) {
-        return this._scripts.hasOwnProperty(name);
+    /* eslint-enable jsdoc/no-undefined-types */
+    ScriptRegistry.prototype.has = function (nameOrType) {
+        if (typeof nameOrType === 'string') {
+            return this._scripts.hasOwnProperty(nameOrType);
+        }
+
+        var scriptName = scriptType.__name;
```
Comment (Maksims) [inline src/script/script-registry.js:206]: Spotted an issue here, it throws error when running: `npm run build:all`.

Comment (aidinabedi) [reply src/script/script-registry.js:206]: @Maksims Fixed issue in https://github.com/playcanvas/engine/pull/1987

**Pull request merged**